src/hotspot/os/aix/os_aix.cpp

```diff
@@ -1159,6 +1158,26 @@ void *os::dll_load(const char *filename, char *ebuf, int ebuflen) {
   }
   return nullptr;
 }
+// Load library named <filename>
+// If filename matches <name>.so, and loading fails, repeat with <name>.a.
+void *os::dll_load(const char *filename, char *ebuf, int ebuflen) {
+  void* result = nullptr;
+  char* const file_path = strdup(filename);
+  char* const pointer_to_dot = strrchr(file_path, '.');
+  const char old_extension[] = ".so";
+  const char new_extension[] = ".a";
+  STATIC_ASSERT(sizeof(old_extension) >= sizeof(new_extension));
+  // First try to load the existing file.
+  result = dll_load_library(filename, ebuf, ebuflen);
+  // If the load fails,we try to reload by changing the extension to .a for .so files only.
+  // Shared object in .so format dont have braces, hence they get removed for archives with members.
+  if (result == nullptr && pointer_to_dot != nullptr && strcmp(pointer_to_dot, old_extension) == 0) {
+      snprintf(pointer_to_dot, sizeof(old_extension), "%s", new_extension);
+      result = dll_load_library(file_path, ebuf, ebuflen);
```

Comment:
You should have adapted the indentation, too.

Reply:
Which indentation ?

Reply:
oh! to reduce the spaces

Reply:
@TheRealMDoerr adopted